src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -398,6 +398,25 @@ public void deleteControlledResource(
     }
   }
 
+  /**
+   * List all controlled resources ID in Sam this user has access to. This builds an aggregated list
+   * across the four types of controlled resources.
+   */
+  @Traced
+  public List<UUID> listControlledResourceIds(
+      AuthenticatedUserRequest userReq, String controlledResourceName) {
```

Comment:
I'm a little confused by the primitive String controlledResourceName and UUID return value. What kind of "name" are we storing in Sam? What kind of UUID, and how does it relate to the ResourceDao, does this return?
Bigger scope than this PR, but I think it would be very useful to strongly type our UUIDs and String "types" of data. For UUID, I know we have at least a "WorkspaceId" and a "ResourceId".  The Sam service also has "resource" as a part of its terminology and API, see SamService#isAuthorized: It's unclear to me what a `String resourceId` represent.